pkg/client/cache/listwatch.go

```diff
@@ -34,22 +35,30 @@ type ListWatch struct {
 
 // ListWatch knows how to list and watch a set of apiserver resources.
 func (lw *ListWatch) List() (runtime.Object, error) {
-	return lw.Client.
-		Get().
-		Namespace(lw.Namespace).
-		Resource(lw.Resource).
-		SelectorParam("fields", lw.FieldSelector).
-		Do().
-		Get()
+	if lw.Client == nil {
```

Comment:
Is the call stack different after that commit?  It looks like Eric guarded that call stack against a nil KubeClient before the call to NewReflector: https://github.com/GoogleCloudPlatform/kubernetes/blob/98bdd3f7c4adf22a1c20d290ce5fd30a11880f91/pkg/kubelet/kubelet.go#L97.